### ISSUE TITLE
How to use cucumber-messages

### ISSUE THREAD

**Issue by xuanzhaopeng** [cucumber/cucumber-js#1288]
Refer to the doc https://github.com/cucumber/cucumber-js/blob/master/docs/cli.md#Formats

It mentioned `message` format,  when I do `--format message`,  it throw an error `Cannot find module /.../message`

I'd like to ask do we have an example about how to use message format? Since currently there is no proper way to generate report **when run scenario in parallel.** 

Thanks for your help

**Comment by davidjgoss** [comment]
There should be a 7.0.0 out quite soon, there are a few things left to fix up.

The current docs can be found here https://github.com/cucumber/cucumber-js/tree/6.x

**Comment by aslakhellesoy** [comment]
@davidjgoss Before we make a release I'd really like to verify that the messages produced by the message formatter can be consumed by tools like [json-formatter](https://github.com/cucumber/cucumber/tree/master/json-formatter) and [html-formatter](https://github.com/cucumber/cucumber/tree/master/html-formatter).

The [compatibility-kit](https://github.com/cucumber/cucumber/tree/master/compatibility-kit) aims to make this a little easier. The [run-formatter](https://github.com/cucumber/cucumber/blob/master/compatibility-kit/scripts/run-formatter) script does this.

**Comment by davidjgoss** [comment]
@aslakhellesoy yeah, I’ve been meaning to have a play with that anyway so will do it soon.

Also based on call from the other day we should change the message formatter to output NDJSON rather than binary?

**Comment by aslakhellesoy** [comment]
Yes! We’re not in a high throughput low latency environment so the benefits of json (human readable) outweighs the marginal benefits of binary.

**Comment by aslakhellesoy** [comment]
Also, I’m releasing the whole monorepo today and tomorrow so there will be some major releases to upgrade

**Comment by xuanzhaopeng** [comment]
Hey @aslakhellesoy  @davidjgoss , I built local version from master branch. I rewrite allure-cucumberjs to consume envelope , and build the allure report based on `this.eventDataCollector`.

In non-parallel mode, It can:
1) consuming envelop success
2) inject variable into World

In parallel mode,  it can:
1) consuming envelop success
2) Cannot inject variable to World. I'm assuming it's because Formmater is working on `Master`, instead of `Slave`.  

May I ask how could we inject variable to World in parallel mode from customized Formmater?

Thanks

**Comment by xuanzhaopeng** [comment]
In other words,  Formmater cannot identify which slave send the message back, and the SLAVE_ID only exposed to each slave,  Is it possible on each envelope message,  that we add slaveId?

**Comment by charlierudolph** [comment]
> There should be a 7.0.0 out quite soon, there are a few things left to fix up.

@davidjgoss you please create an issue where we start tracking what tasks we want to complete before 7.0.0?

> May I ask how could we inject variable to World in parallel mode from customized Formmater?

@xuanzhaopeng I don't see the reason a good reason should allow this. To me formatters should not effect execution. What is the reason you need this? Maybe we could solve that in a different way.

> In other words, Formmater cannot identify which slave send the message back, and the SLAVE_ID only exposed to each slave, Is it possible to add SLAVE_ID on TestCaseStarted message

Why does the formatter want to know what slave ran each test case? Adding the slave to the message is certainly possible, but I'd like to understand why you want it.

**Comment by xuanzhaopeng** [comment]
Hey @charlierudolph , for 1st point,  I think it's very specific allure related issue, maybe we could discuss it in separated thread.

For adding slave id into message,  is because: 
When run test in parallel,  Allure report can show the time slot like this(maybe also other reporter consumers):
![image](https://user-images.githubusercontent.com/5563244/74548594-6d6b1880-4f4e-11ea-9ca1-cc78fcdaa81b.png)

So it can be achieved by `allure.addLabel(LabelName.THREAD, slaveId)`, so if we could get Slave ID from `TestCaseStarted` message,  this information could be easily used by allure,  and show exact execution duration for each slave. 

I made an simple test locally by using exist message proto(testCaseStarted.platform.cpu)  
```
                testCaseStarted: {
                    attempt,
                    testCaseId: this.testCaseId,
                    id: this.currentTestCaseStartedId,
                    platform: {
                        cpu: process.env.CUCUMBER_SLAVE_ID || '-1'
                    }
                },
```
refer to code
https://github.com/cucumber/cucumber-js/blob/master/src/runtime/pickle_runner.ts#L269

And it works perfect in allure

**Comment by xuanzhaopeng** [comment]
Hey @charlierudolph ,  do you think we can put slave number inside of testCaseStarted Message in 7.0.0 release?

**Comment by davidjgoss** [comment]
Closing this as the original ask - documentation for message - was answered.

Regarding new elements to the messages, if that's still relevant please raise in the main repo https://github.com/cucumber/common - thanks!
